sos/plugins/samba.py

```diff
@@ -23,16 +23,32 @@ class Samba(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
     profiles = ('services',)
 
     def setup(self):
-        self.add_copy_spec([
-            "/etc/samba/smb.conf",
-            "/etc/samba/lmhosts",
-            "/var/log/samba/log.smbd",
-            "/var/log/samba/log.nmbd"
-        ])
+        self.limit = self.get_option("log_size")
+
+        self.add_copy_spec_limit("/etc/samba/smb.conf",
+                                 sizelimit=self.limit)
+        self.add_copy_spec_limit("/etc/samba/lmhosts",
+                                 sizelimit=self.limit)
+        self.add_copy_spec_limit("/var/log/samba/log.smbd",
+                                 sizelimit=self.limit)
+        self.add_copy_spec_limit("/var/log/samba/log.nmbd",
+                                 sizelimit=self.limit)
+        self.add_copy_spec_limit("/var/log/samba/log.winbindd",
+                                 sizelimit=self.limit)
+        self.add_copy_spec_limit("/var/log/samba/log.winbindd-idmap",
+                                 sizelimit=self.limit)
+        self.add_copy_spec_limit("/var/log/samba/log.winbindd-dc-connet",
+                                 sizelimit=self.limit)
+        self.add_copy_spec_limit("/var/log/samba/log.wb-*",
+                                 sizelimit=self.limit)
+
+        if self.get_option("all_logs"):
```

Comment:
Wouldn't this mean that we get two copies of the above mentioned logs with --all-logs option?

What about having something like below?

if not self.get_option("all_logs"):
    limit = self.get_option("log_size")
else:
    limit = 0

if limit:
    self.add_copy_spec_limit("/var/log/samba/log.smbd",
                                             sizelimit=self.limit)
    self.add_copy_spec_limit("/var/log/samba/log.nmbd",
                                             sizelimit=self.limit)
    self.add_copy_spec_limit("/var/log/samba/log.winbindd",
                                             sizelimit=self.limit)
    self.add_copy_spec_limit("/var/log/samba/log.winbindd-idmap",
                                             sizelimit=self.limit)
    self.add_copy_spec_limit("/var/log/samba/log.winbindd-dc-connet",
                                             sizelimit=self.limit)
    self.add_copy_spec_limit("/var/log/samba/log.wb-*",
                                             sizelimit=self.limit)
else:
            self.add_copy_spec("/var/log/samba")